sklearn/cluster/_agglomerative.py

```diff
@@ -455,6 +455,11 @@ def linkage_tree(X, connectivity=None, n_clusters=None, linkage='complete',
             # data, provide as first argument an ndarray of the shape returned
             # by pdist: it is a flat array containing the upper triangular of
             # the distance matrix.
+            if len(X.shape) != 2 or X.shape[0] != X.shape[1]:
+                raise ValueError(
+                    'Matrix should be square, as returned by pdist. '
```

Comment:
pairwise_distances would be more familiar to our users than pdist

Reply:
Ok, thanks. Let me change the Error Message.